In [2]:
import pandas as pd
import numpy as np
from datetime import datetime
import uuid
import requests
import json
import random
import time
import boto3

In [105]:
grocery_items = ['tropical fruit', 'whole milk', 'pip fruit', 'other vegetables',
       'rolls/buns', 'pot plants', 'citrus fruit', 'beef', 'frankfurter',
       'chicken', 'butter', 'fruit/vegetable juice',
       'packaged fruit/vegetables', 'chocolate', 'specialty bar',
       'butter milk', 'bottled water', 'yogurt', 'sausage', 'brown bread',
       'hamburger meat', 'root vegetables', 'pork', 'pastry',
       'canned beer', 'berries', 'coffee', 'misc. beverages', 'ham',
       'turkey', 'curd cheese', 'red/blush wine',
       'frozen potato products', 'flour', 'sugar', 'frozen meals',
       'herbs', 'soda', 'detergent', 'grapes', 'processed cheese', 'fish',
       'sparkling wine', 'newspapers', 'curd', 'pasta', 'popcorn',
       'finished products', 'beverages', 'bottled beer', 'dessert',
       'dog food', 'specialty chocolate', 'condensed milk', 'cleaner',
       'white wine', 'meat', 'ice cream', 'hard cheese', 'cream cheese ',
       'liquor', 'pickled vegetables', 'liquor (appetizer)', 'UHT-milk',
       'candy', 'onions', 'hair spray', 'photo/film', 'domestic eggs',
       'margarine', 'shopping bags', 'salt', 'oil', 'whipped/sour cream',
       'frozen vegetables', 'sliced cheese', 'dish cleaner',
       'baking powder', 'specialty cheese', 'salty snack',
       'Instant food products', 'pet care', 'white bread',
       'female sanitary products', 'cling film/bags', 'soap',
       'frozen chicken', 'house keeping products', 'spread cheese',
       'decalcifier', 'frozen dessert', 'vinegar', 'nuts/prunes',
       'potato products', 'frozen fish', 'hygiene articles',
       'artif. sweetener', 'light bulbs', 'canned vegetables',
       'chewing gum', 'canned fish', 'cookware', 'semi-finished bread',
       'cat food', 'bathroom cleaner', 'prosecco', 'liver loaf',
       'zwieback', 'canned fruit', 'frozen fruits', 'brandy',
       'baby cosmetics', 'spices', 'napkins', 'waffles', 'sauces', 'rum',
       'chocolate marshmallow', 'long life bakery product', 'bags',
       'sweet spreads', 'soups', 'mustard', 'specialty fat',
       'instant coffee', 'snack products', 'organic sausage',
       'soft cheese', 'mayonnaise', 'dental care', 'roll products ',
       'kitchen towels', 'flower soil/fertilizer', 'cereals',
       'meat spreads', 'dishes', 'male cosmetics', 'candles', 'whisky',
       'tidbits', 'cooking chocolate', 'seasonal products', 'liqueur',
       'abrasive cleaner', 'syrup', 'ketchup', 'cream', 'skin care',
       'rubbing alcohol', 'nut snack', 'cocoa drinks', 'softener',
       'organic products', 'cake bar', 'honey', 'jam', 'kitchen utensil',
       'flower (seeds)', 'rice', 'tea', 'salad dressing',
       'specialty vegetables', 'pudding powder', 'ready soups',
       'make up remover', 'toilet cleaner', 'preservation products']
grocery_items_count = len(grocery_items)

In [86]:
order_status = ['IN_PROGRESS'] * 48 + ['COMPLETE'] * 45 + ['DELAYED'] * 3

In [88]:
number_of_results = 20
r = requests.get('https://randomuser.me/api/?exc=login&results=' + str(number_of_results))
data = r.json()["results"]
random_items_count = int(random.uniform(1, 10))

In [91]:
data

[{'gender': 'female',
  'name': {'title': 'Mrs', 'first': 'Noor', 'last': 'Pilskog'},
  'location': {'street': {'number': 1994, 'name': 'Eventyrveien'},
   'city': 'Nordvågen',
   'state': 'Hedmark',
   'country': 'Norway',
   'postcode': '5948',
   'coordinates': {'latitude': '53.8415', 'longitude': '136.5427'},
   'timezone': {'offset': '+11:00',
    'description': 'Magadan, Solomon Islands, New Caledonia'}},
  'email': 'noor.pilskog@example.com',
  'dob': {'date': '1972-04-30T23:26:25.427Z', 'age': 49},
  'registered': {'date': '2011-11-05T22:25:26.155Z', 'age': 10},
  'phone': '75876158',
  'cell': '99931959',
  'id': {'name': 'FN', 'value': '30047232607'},
  'picture': {'large': 'https://randomuser.me/api/portraits/women/68.jpg',
   'medium': 'https://randomuser.me/api/portraits/med/women/68.jpg',
   'thumbnail': 'https://randomuser.me/api/portraits/thumb/women/68.jpg'},
  'nat': 'NO'},
 {'gender': 'female',
  'name': {'title': 'Ms', 'first': 'Leyla', 'last': 'Schorn'},
  'locatio

In [89]:
client = boto3.client('kinesis',
                        region_name='us-east-1',
                        aws_access_key_id='AKIAYKQTFN3DBO6PCFUW',
                        aws_secret_access_key='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')
stream_name = 'my-second-stream'

In [67]:
print_i = 0
while True:
        random_user_index = int(random.uniform(0, (number_of_results - 1)))
        random_user = data[random_user_index]
        partition_key = random_user["email"]
        random_user["order_number"] = str(uuid.uuid4())
        random_user["order_status"] = random.choice(order_status)
        total_cost = random.uniform(1.0, 199.99)
        costs = [str(total_cost)] * 95 + [str(random.uniform(1000.0, 5000.99))] * 5
        total_cost = random.choice(costs)
        total_cost = round(total_cost, 2)
        random_user["total_cost"] = total_cost
        items = []
        for i in range(0, random_items_count):
            random_grocery_item_index = int(random.uniform(0, (grocery_items_count - 1)))
            items.append(grocery_items[random_grocery_item_index])
        random_user["items"] = items
        order = json.dumps(random_user)
        try:
            client.put_record(
                StreamName=stream_name,
                Data=order,
                PartitionKey=partition_key)
            if(print_i == 0):
                print("Adding orders to kinesis stream named %s..." % stream_name)
            if((print_i % 100) == 0):
                print("Still adding orders to kinesis stream named %s..." % stream_name)
        except ClientError as e:
                print("Unexpected error: %s" % e)
        time.sleep(random.uniform(0, 1))
        print_i = print_i + 1

Adding orders to kinesis stream named my-second-stream...
Still adding orders to kinesis stream named my-second-stream...
Still adding orders to kinesis stream named my-second-stream...
Still adding orders to kinesis stream named my-second-stream...
Still adding orders to kinesis stream named my-second-stream...
Still adding orders to kinesis stream named my-second-stream...
Still adding orders to kinesis stream named my-second-stream...
Still adding orders to kinesis stream named my-second-stream...
Still adding orders to kinesis stream named my-second-stream...
Still adding orders to kinesis stream named my-second-stream...


KeyboardInterrupt: 

In [62]:
print_i = 0

random_user_index = int(random.uniform(0, (number_of_results - 1)))
random_user = data[random_user_index]
partition_key = random_user["email"]
random_user["order_number"] = str(uuid.uuid4())
random_user["order_status"] = random.choice(order_status)
total_cost = random.uniform(1.0, 199.99)
total_cost = round(total_cost, 2)
random_user["total_cost"] = total_cost
items = []
for i in range(0, random_items_count):
    random_grocery_item_index = int(random.uniform(0, (grocery_items_count - 1)))
    items.append(grocery_items[random_grocery_item_index])
random_user["items"] = items
order = json.dumps(random_user)
try:
    client.put_record(
        StreamName=stream_name,
        Data=order,
        PartitionKey=partition_key)
    if(print_i == 0):
        print("Adding orders to kinesis stream named %s..." % stream_name)
    if((print_i % 100) == 0):
        print("Still adding orders to kinesis stream named %s..." % stream_name)
except ClientError as e:
        print("Unexpected error: %s" % e)
time.sleep(random.uniform(0, 1))
print_i = print_i + 1
print(order)

Adding orders to kinesis stream named my-stream...
Still adding orders to kinesis stream named my-stream...
{"gender": "female", "name": {"title": "Mrs", "first": "Taylor", "last": "Ryan"}, "location": {"street": {"number": 3281, "name": "W Campbell Ave"}, "city": "New Orleans", "state": "Arizona", "country": "United States", "postcode": 54395, "coordinates": {"latitude": "7.0426", "longitude": "-49.4980"}, "timezone": {"offset": "+5:30", "description": "Bombay, Calcutta, Madras, New Delhi"}}, "email": "taylor.ryan@example.com", "dob": {"date": "1970-01-16T21:35:43.182Z", "age": 51}, "registered": {"date": "2002-10-20T00:40:08.316Z", "age": 19}, "phone": "(759)-788-0862", "cell": "(815)-634-2249", "id": {"name": "SSN", "value": "721-92-4190"}, "picture": {"large": "https://randomuser.me/api/portraits/women/34.jpg", "medium": "https://randomuser.me/api/portraits/med/women/34.jpg", "thumbnail": "https://randomuser.me/api/portraits/thumb/women/34.jpg"}, "nat": "US", "order_number": "67141

In [ ]:
# Kinesis Data Analytics SQL Query
# CREATE OR REPLACE STREAM "DESTINATION_USER_DATA" (
#     order_number VARCHAR(56), 
#     email VARCHAR(56), 
#     order_status VARCHAR(16)
# );
# CREATE OR REPLACE PUMP "STREAM_PUMP" AS INSERT INTO "DESTINATION_USER_DATA"

# SELECT STREAM "order_number", "email", "order_status"
# FROM "SOURCE_SQL_STREAM_001"
# WHERE "order_status" = 'DELAYED'

In [109]:
client = boto3.client('kinesis',
                        region_name='us-east-1',
                        aws_access_key_id='AKIAYTTPC3LCRAOQKN7Z',
                        aws_secret_access_key='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')
stream_name = 'my-stream'

In [113]:

order = {}
order["order_number"] = str(uuid.uuid4())
order["order_status"] = random.choice(order_status)

items = []
for i in range(0, random_items_count):
    random_grocery_item_index = int(random.uniform(0, (grocery_items_count - 1)))
    items.append(grocery_items[random_grocery_item_index])
order["items"] = items

costs = [str(random.uniform(1.0, 199.99))] * 95 + [str(random.uniform(1000.0, 5000.99))] * 5
total_cost = random.choice(costs)
total_cost = round(float(total_cost), 2)
order["total_cost"] = total_cost

user_ids = ['uid001', 'uid002']
order["user_id"] = random.choice(user_ids)


partition_key = order["user_id"]

order = json.dumps(order)
print(order)

try:
    client.put_record(
        StreamName=stream_name,
        Data=order,
        PartitionKey=partition_key)
    if(print_i == 0):
        print("Adding orders to kinesis stream named %s..." % stream_name)
    if((print_i % 100) == 0):
        print("Still adding orders to kinesis stream named %s..." % stream_name)
except ClientError as e:
        print("Unexpected error: %s" % e)
time.sleep(random.uniform(0, 1))


{"order_number": "e299b0e6-485e-4f76-9cef-683805848bbc", "order_status": "COMPLETE", "items": ["baby cosmetics", "nut snack", "tea", "canned fruit", "roll products ", "chocolate marshmallow", "rice"], "total_cost": 3439.81, "user_id": "uid002"}


In [ ]:
print_i = 0
while True:
    order = {}
    order["order_number"] = str(uuid.uuid4())
    order["order_status"] = random.choice(order_status)

    items = []
    for i in range(0, random_items_count):
        random_grocery_item_index = int(random.uniform(0, (grocery_items_count - 1)))
        items.append(grocery_items[random_grocery_item_index])
    order["items"] = items

    costs = [str(random.uniform(1.0, 199.99))] * 95 + [str(random.uniform(1000.0, 5000.99))] * 5
    total_cost = random.choice(costs)
    total_cost = round(float(total_cost), 2)
    order["total_cost"] = total_cost

    user_ids = ['uid001', 'uid002']
    order["user_id"] = random.choice(user_ids)
    
    partition_key = order["user_id"]

    order = json.dumps(order)
    

    try:
        client.put_record(
            StreamName=stream_name,
            Data=order,
            PartitionKey=partition_key)
        if(print_i == 0):
            print("Adding orders to kinesis stream named %s..." % stream_name)
        if((print_i % 100) == 0):
            print("Still adding orders to kinesis stream named %s..." % stream_name)
    except ClientError as e:
            print("Unexpected error: %s" % e)
    time.sleep(random.uniform(0, 1))
    print_i = print_i + 1

In [17]:
number_of_results = 500
r = requests.get('https://randomuser.me/api/?exc=login&nat=us,dk,fr,gb&results=' + str(number_of_results))
data = r.json()["results"]

In [24]:
users = []
for i in range(0, number_of_results):
    curr_user = data[i]
    user = {}
    user["user_id"] = 'uid_' + str(i + 1)
    user["email"] = curr_user["email"]
    user["first_name"] = curr_user["name"]["first"]
    user["last_name"] = curr_user["name"]["last"]
    user["street_number"] = curr_user["location"]["street"]["number"]
    user["stree_name"] = curr_user["location"]["street"]["name"]
    user["city"] = curr_user["location"]["city"]
    user["state"] = curr_user["location"]["state"]
    user["country"] = curr_user["location"]["country"]
    user["postcode"] = curr_user["location"]["postcode"]
    user["latitute"] = curr_user["location"]["coordinates"]["latitude"]
    user["longitude"] = curr_user["location"]["coordinates"]["longitude"]
    users.append(user)

print(users)    
with open('user_json.json', 'w') as outfile:
    json.dump(users, outfile)

[{'user_id': 'uid_1', 'email': 'andrea.olsen@example.com', 'first_name': 'Andrea', 'last_name': 'Olsen', 'street_number': 5216, 'stree_name': 'Nordre Strandvej', 'city': 'Nørre Sundby', 'state': 'Nordjylland', 'country': 'Denmark', 'postcode': 72397, 'latitute': '-8.9240', 'longitude': '110.0525'}, {'user_id': 'uid_2', 'email': 'arnold.hunt@example.com', 'first_name': 'Arnold', 'last_name': 'Hunt', 'street_number': 7893, 'stree_name': 'The Grove', 'city': 'Brighton and Hove', 'state': 'Gloucestershire', 'country': 'United Kingdom', 'postcode': 'EE7V 6ZN', 'latitute': '53.4580', 'longitude': '151.8459'}, {'user_id': 'uid_3', 'email': 'jacob.sorensen@example.com', 'first_name': 'Jacob', 'last_name': 'Sørensen', 'street_number': 4979, 'stree_name': 'Fredericiagade', 'city': 'Viby J.', 'state': 'Hovedstaden', 'country': 'Denmark', 'postcode': 68197, 'latitute': '-25.1934', 'longitude': '149.8577'}, {'user_id': 'uid_4', 'email': 'rasmus.nielsen@example.com', 'first_name': 'Rasmus', 'last_na

{"title": "Mrs", "first": "Taylor", "last": "Ryan"}, "location": {"street": {"number": 3281, "name": "W Campbell Ave"}, "city": "New Orleans", "state": "Arizona", "country": "United States", "postcode": 54395, "coordinates": {"latitude": "7.0426", "longitude": "-49.4980"}